stock_release_channel/models/stock_picking.py

```diff
@@ -15,8 +15,34 @@ class StockPicking(models.Model):
         ondelete="restrict",
         copy=False,
         tracking=True,
+        inverse="_inverse_release_channel_id",
     )
 
+    def _inverse_release_channel_id(self):
+        """When the release channel is modified, update the expected date
+
+        Set new date on all moves in the chain"""
+        for delivery in self:
+            if not delivery.last_release_date:
+                continue
+            if delivery.picking_type_code != "outgoing":
+                continue
+            channel = delivery.release_channel_id
```

Comment:
```suggestion
            channel = delivery.release_channel_id
            if not channel:
                continue
```
Here it is mandatory to have a channel assigned. `_get_expected_date` expects a record

Reply:
fixed